OpenPGL.NET/Field.cs

```diff
@@ -200,15 +210,51 @@ public class VMMDirectionalSettings : DirectionalSettings {
     }
 }
 
+public enum DQTLeafEstimator {
+    RejectionSampling = 0,
+    PerLeaf
+};
+
+public enum DQTSplitMetric {
+    Mean = 0,
+    SecondMoment
+};
+
+public class DQTDirectionalSettings : DirectionalSettings {
+    public DQTLeafEstimator LeafEstimator = DQTLeafEstimator.RejectionSampling;
+    public DQTSplitMetric SplitMetric = DQTSplitMetric.Mean;
+    public float SplitThreshold = 0.01f;
+    public float FootprintFactor = 1;
+    public int MaxLevels = 12;
+    
+    internal override OpenPGL.PGL_DIRECTIONAL_DISTRIBUTION_TYPE GetDistType() {
+        return OpenPGL.PGL_DIRECTIONAL_DISTRIBUTION_TYPE.PGL_DIRECTIONAL_DISTRIBUTION_QUADTREE;
+    }
+
+    internal override void SetArguments(ref OpenPGL.PGLFieldArguments target) {
+        target.directionalDistributionType = GetDistType();
+
+        OpenPGL.PGLDQTFactoryArguments args = new() {
+            leafEstimator = (OpenPGL.PGLDQTLeafEstimator) LeafEstimator,
+            splitMetric = (OpenPGL.PGLDQTSplitMetric) SplitMetric,
+            splitThreshold = SplitThreshold,
+            footprintFactor = FootprintFactor,
+            maxLevels = (uint)int.Max(0, MaxLevels),
+        };
+        Marshal.StructureToPtr<OpenPGL.PGLDQTFactoryArguments>(args, target.directionalDistributionArguments, true);
+    }
+}
+
 public struct FieldSettings {
     public SpatialSettings SpatialSettings { get; init; }
     public DirectionalSettings DirectionalSettings { get; init; }
 
     internal OpenPGL.PGLFieldArguments MakeArguments() {
+        // We require the correct struct and dist type for preallocation done inside pglFieldArgumentsSetDefaults
         OpenPGL.PGLFieldArguments arguments;
         OpenPGL.pglFieldArgumentsSetDefaults(out arguments,
-            OpenPGL.PGL_SPATIAL_STRUCTURE_TYPE.PGL_SPATIAL_STRUCTURE_KDTREE,
-            OpenPGL.PGL_DIRECTIONAL_DISTRIBUTION_TYPE.PGL_DIRECTIONAL_DISTRIBUTION_PARALLAX_AWARE_VMM);
+            SpatialSettings.GetStructType(),

```

Comment:
If you want to do that, the properties above need a non-null default. Else this will crash if the user tries to run with default settings.